packages/@ember/-internals/owner/index.ts

```diff
@@ -99,5 +101,27 @@ export function getOwner(object: any): Owner {
   @public
 */
 export function setOwner(object: any, owner: Owner): void {
+  if (DEBUG) {
+    let publicApiType = owner as any;
+
+    assert(`Expected owner to be truthy`, publicApiType);
+    assert(
+      'Passed owner must have a function, `lookup`',
+      typeof publicApiType.lookup === 'function'
+    );
+    assert(
+      'Passed owner must have a function, `factoryFor`',
+      typeof publicApiType.factoryFor === 'function'
+    );
+    assert(
+      'Passed owner must have a function, `register`',
+      typeof publicApiType.register === 'function'
+    );
+    assert(
+      'Passed owner must have a function, `hasRegistration`',
+      typeof publicApiType.hasRegistration === 'function'
+    );
+  }
```

Comment:
`buildChildEngineInstance` is omitted because it _seemed_ like private api _even though_ the type declares it as required.
idk if this is the right call, but tests were failing with that check, so that leads me to believe that either tests are incorrectly building an owner, or `buildChildEngineInstance` needs to be marked as optional

Reply:
looks like there are tests passing `undefined` or `null` to `setOwner` 🤔